awx/main/management/commands/revoke_tokens.py

```diff
@@ -0,0 +1,34 @@
+# Django
+from django.core.management.base import BaseCommand, CommandError
+from django.contrib.auth.models import User
+from django.core.exceptions import ObjectDoesNotExist
+
+# AWX
+from awx.main.models.oauth import OAuth2AccessToken
+from oauth2_provider.models import RefreshToken
+
+def revoke_tokens(token_list):
+    for token in token_list:
+            token.revoke()
+
+class Command(BaseCommand):
+    """Command that revokes OAuth2 tokens and refresh tokens."""
+    help='Revokes OAuth2 tokens and refresh tokens.'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--user', dest='user', type=str)
+        parser.add_argument('--revoke_refresh', dest='revoke_refresh', action='store_true')
+
+    def handle(self, *args, **options):
+        if not options['user']:
+            if options['revoke_refresh']:
+                revoke_tokens(RefreshToken.objects.all())
+            revoke_tokens(OAuth2AccessToken.objects.all())
+        else:
+            try:
+                user = User.objects.get(username=options['user'])
+            except ObjectDoesNotExist:
+                raise CommandError('The user does not exist.')
+            if options['revoke_refresh']:
+                revoke_tokens(RefreshToken.objects.filter(user=user))
+            revoke_tokens(user.main_oauth2accesstoken.filter(user=user))
```

Comment:
It looks like when you call `token.revoke()` on a RefreshToken, django-oauth-toolkit (DOT) attempts to delete the related AccessToken ([here](https://github.com/jazzband/django-oauth-toolkit/blob/master/oauth2_provider/models.py#L380).  If the related AccessToken doesn't exist (which is possible as of DOT 1.1.2), we will run into the following strace.  

```
[root@awx /]# awx-manage revoke_tokens --user gandalf --revoke_refresh
Traceback (most recent call last):
  File "/usr/local/bin/awx-manage", line 9, in <module>
    load_entry_point('awx', 'console_scripts', 'awx-manage')()
  File "/awx_devel/awx/__init__.py", line 116, in manage
    execute_from_command_line(sys.argv)
  File "/venv/awx/lib/python2.7/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/venv/awx/lib/python2.7/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/venv/awx/lib/python2.7/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/venv/awx/lib/python2.7/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/awx_devel/awx/main/management/commands/revoke_tokens.py", line 33, in handle
    revoke_tokens(RefreshToken.objects.filter(user=user))
  File "/awx_devel/awx/main/management/commands/revoke_tokens.py", line 12, in revoke_tokens
    token.revoke()
  File "/venv/awx/lib/python2.7/site-packages/oauth2_provider/models.py", line 371, in revoke
    access_token_model.objects.get(id=self.access_token_id).revoke()
  File "/venv/awx/lib/python2.7/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/venv/awx/lib/python2.7/site-packages/django/db/models/query.py", line 380, in get
    self.model._meta.object_name
awx.main.models.oauth.DoesNotExist: OAuth2AccessToken matching query does not exist.
```

One proposed workaround until this is fixed in DOT, is to amend your `revoke_token()` function to just delete the AWX token resources.